oioswift/proxy/controllers/obj.py

```diff
@@ -663,6 +664,10 @@ def DELETE(self, req):
     def _delete_object(self, req):
         storage = self.app.storage
         oio_headers = {'X-oio-req-id': self.trans_id}
+        # Allow swift3 to delete things from the multipart container
+        if (req.environ.get('swift.source') == 'S3' and
+                self.container_name.endswith(MULTIUPLOAD_SUFFIX)):
```

Comment:
IMO it will not work with `Container Hierachy` or `regexcontainer`, maybe add a test regarding in `oio-bucket-name` ?